tests/api.c

```diff
@@ -17839,7 +17838,31 @@ static void test_wolfSSL_X509_NAME(void)
     printf(resultFmt, passed);
     #endif /* defined(OPENSSL_EXTRA) && !defined(NO_DES3) */
 }
+static void test_wolfSSL_X509_subject_name_hash(void)
+{
+#if defined(OPENSSL_EXTRA) && !defined(NO_CERTS) && !defined(NO_FILESYSTEM) \
+    && defined(WOLFSSL_CERT_GEN) && defined(WOLFSSL_CERT_REQ)
+
+    X509* x509;
+    X509_NAME* subjectName = NULL;
+    unsigned long ret = 0;
 
+    printf(testingFmt, "wolfSSL_X509_subject_name_hash()");
+
+    AssertNotNull(x509 = wolfSSL_X509_load_certificate_file(cliCertFile,
+                SSL_FILETYPE_PEM));
+
+    AssertNotNull(subjectName = wolfSSL_X509_get_subject_name(x509));
+
+    ret = wolfSSL_X509_subject_name_hash(x509);
+
+    AssertIntNE(ret, WOLFSSL_FATAL_ERROR);
```

Comment:
with ret being an unsigned value it can never be -1, also the return value should be a constant known value that can be checked here